zeppelin-server/src/main/java/org/apache/zeppelin/realm/LdapRealm.java

```diff
@@ -204,6 +213,34 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
     }
   }
 
+  protected void onInit() {
+    super.onInit();
+    if (!org.apache.commons.lang.StringUtils.isEmpty(this.hadoopSecurityCredentialPath)
+        && getContextFactory() != null) {
+      LdapContextFactory defaultFactory = getContextFactory();
+      ((JndiLdapContextFactory) defaultFactory).setSystemPassword(getSystemPassword());
+      setContextFactory(defaultFactory);
+    }
+  }
+
+  private String getSystemPassword() {
+    String password = "";
+    try {
+      Configuration configuration = new Configuration();
+      configuration.set(CredentialProviderFactory.CREDENTIAL_PROVIDER_PATH,
+          this.hadoopSecurityCredentialPath);
+      CredentialProvider provider = CredentialProviderFactory.getProviders(configuration).get(0);
+      CredentialProvider.CredentialEntry credEntry = provider.getCredentialEntry(
+          keystorePass);
+      if (credEntry != null) {
+        password = new String(credEntry.getCredential());
+      }
+    } catch (Exception e) {
+      log.debug("ignored error from getting credential entry from keystore", e);
```

Comment:
Right, fair point let me log this exception.

Reply:
I think it is better to fail zeppelin server instead of logging the exception.  In general, I think for any invalid configuration, we should fail the zeppelin server. Otherwise user may be confused that some features doesn't' work due to invalid configuration unless they check the log.

Reply:
Sure now Zeppelin server will fail to start and throw "HTTP ERROR: 503" in case of invalid configuration related to hadoopSecurityCredentialPath for both `ActiveDirectoryGroupRealm` and `LdapRealm`